libpromises/evalfunction.c

```diff
@@ -2931,7 +2931,10 @@ JsonElement *DefaultTemplateData(const EvalContext *ctx, const char *wantbundle)
             if (NULL != scope_obj)
             {
                 char *lval_key = VarRefToString(var->ref, false);
-                JsonObjectAppendElement(scope_obj, lval_key, RvalToJson(var->rval));
+                if (NULL == strchr(lval_key, '#')) // don't collect mangled refs
+                {
+                    JsonObjectAppendElement(scope_obj, lval_key, RvalToJson(var->rval));
+                }
```

Comment:
`this#promise_filename` shows up in the bundle's variables (claims its scope is the bundle).  I don't know why it happens since I didn't write the variable-gathering code, but since we can't have variables with `#` in the name, the fix is good regardless of the underlying issue.